Web view: new prop: Open links externally

sergiuszwoznicki opened the pull request:
Web view widget: added new property: "Open links externally".
If true, links embedded in source document will be opened in external browser instead of inside this widget.

diego-antonelli commented on packages-native/web-view/src/WebView.tsx:
```diff
@@ -28,6 +28,7 @@ export class WebView extends Component<Props> {
 
         return (
             <View style={this.styles.container}>
+                <Text style={this.styles.errorText}>ohoho</Text>
```
Think you need to change `ohoho` to a proper error..

sergiuszwoznicki commented:
> Can you add a full description with why this is necessary and how it works?
Well, not necessary, just a nice-to-have. In our app, we had a requirement to open nested links in external browser insted of the web view instance. The solution is based on the 2nd answer here: https://stackoverflow.com/questions/35531679/react-native-open-links-in-browser

> I would also say you need to bump the version of this widget in package.json and package.xml.
Done.

> Think you need to change ohoho to a proper error.
Sorry, stupid mistake, a temp debug left; fixed: removed it.

diego-antonelli commented:
@sergiuszwoznicki can you please fix the linting for Webview.spec ? The build is failing because of it.

sergiuszwoznicki commented:
Yep, done.

Sergiusz Woznicki
Senior Software Developer

From: Diego Antonelli <notifications@github.com>
Sent: Tuesday, July 14, 2020 11:25 AM
To: mendix/widgets-resources <widgets-resources@noreply.github.com>
Cc: Sergiusz Woznicki <swoznicki@objectivity.co.uk>; Mention <mention@noreply.github.com>
Subject: Re: [mendix/widgets-resources] Web view: new prop: Open links externally (#238)


@sergiuszwoznicki<https://github.com/sergiuszwoznicki> can you please fix the linting for Webview.spec ? The build is failing because of it.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/mendix/widgets-resources/pull/238#issuecomment-658074740>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGD7MEQU3BVUFPY337E5COTR3QP5FANCNFSM4OZIVBWQ>.

diego-antonelli commented on packages-native/web-view/src/WebView.tsx:
```diff
@@ -38,6 +38,16 @@ export class WebView extends Component<Props> {
                     onLoad={this.onLoadHandler}
                     onError={this.onErrorHandler}
                     userAgent={this.props.userAgent}
+                    onShouldStartLoadWithRequest={event => {
```
event is just being used to check url, can you instead of use event, use `({url}) => ` ?

sergiuszwoznicki replied:
Good point, cleaner code, done.

diego-antonelli commented:
@leonardomendix can you proceed with testing before merge?

diego-antonelli commented:
This will be available in the next release of Native Mobile Resources
